pkg/table/column.go

```diff
@@ -624,7 +642,7 @@ func getColDefaultValue(ctx expression.BuildContext, col *model.ColumnInfo, defa
 }
 
 func getColDefaultValueFromNil(ctx expression.BuildContext, col *model.ColumnInfo, args *getColOriginDefaultValue) (types.Datum, error) {
-	if !mysql.HasNotNullFlag(col.GetFlag()) && !mysql.HasNoDefaultValueFlag(col.GetFlag()) {
+	if !mysql.HasNotNullFlag(col.GetFlag()) {
```

Comment:
seems line 633 always return true after this change

Reply:
nice catch, done.